Adding reports data table pages

noahmanger opened the pull request:
This creates pages for presidential, house and senate, pac and party, and ie only reports data table pages. By default, these pages only show the most recent, unamended version of the report, and are sorted by coverage date. 

I have a separate issue started here with API changes that this will depend on: https://github.com/18F/openFEC/issues/1804

Sharing this early as a WIP for visibility into API needs.

Resolves https://github.com/18F/openFEC/issues/1790
## Update on this:
- This now includes all the filters and fields for reports pages
- I updated the filings page:
  - No longer includes financial columns
  - No longer includes "report type" filters, since those are available on reports pages
  - Report rows on the filings table share the same details panel template as on the reports pages, so now Form 5s include a details panel
- I added a new "range filter" for financial ranges, and replaced all instances of min and max financial fields with this new range macro (but no styles yet)

This should be all ready to go. Finishing up a few things in a related PR in fec-style, but not a blocker.

Commit:
Adding reports data table pages

Commit:
Adding report-specific details templates

Commit:
Adding default null value to currency column

Now, null values register as $0

Commit:
Revert "Adding default null value to currency column"

This reverts commit d7176ca24259d11af6749d62a9701d0a9ba895d7.

Commit:
Adding receipt_date column

Commit:
Adding amendment status & filters

Commit:
Switching source links to reports

Commit:
Adding PCEs and IEs

Commit:
Re-arranging candidate details panel

Commit:
Renaming filter panel

Commit:
Adding pages to filings table

Commit:
Adding coverage beginning date

Commit:
Adding committee name column

Commit:
Adding receipt date column and more filters

Commit:
Renaming filter param

cc @jontours

Commit:
Adding amount range filters

Commit:
Removing receipt-date columns

Commit:
Adding committee type filter

Commit:
Adding authorized candidate id filter

Commit:
Merge branch 'develop' of github.com:18F/openFEC-web-app into feature/reports-pages

Conflicts:
	openfecwebapp/templates/partials/candidate/financial-summary.html

Commit:
Replacing all currency filters with range filters

Commit:
Using same report templates for filings

ccostino commented on openfecwebapp/templates/datatable.html:
```diff
@@ -48,7 +47,14 @@
+{% if form_type %}
+<script>
+var context = {
+  formType: '{{ form_type }}',
```
Is the idea to have both of these properties be in the `context` object if the conditions allow for it?  It doesn't look like that'll be the case here, and I wonder if there may be a possibility of both conditional checks being true.  If that were to happen, `context` would be overwritten and only contain what is in the last `if` statement here.

If this is not the case, the extra comma at the end might be an issue, I thought JS complained about that after the last item in a literal array or object definition.

noahmanger replied:
So yeah, that's the idea: the `office` variable is only being passed through on candidate office pages, and the `form_type` variable is only being passed through on reports pages. But that's a good point: it's definitely brittle and liable to cause problems if we ever end up trying to pass both. I'll think of a better way to do it.

noahmanger replied:
Ok, came up with a solution for this.

ccostino commented on openfecwebapp/templates/macros/filters/range.html:
```diff
@@ -0,0 +1,16 @@
+  <legend class="label">{{ label }}</legend>
+  <div class="range">
+    <div class="range__input js-filter">
+      <label>From</label>
```
Note that you can add a `for` attribute in a `label` element and set it to match the `id` of the input element.  This helps with [accessibility](https://pages.18f.gov/accessibility/forms/) as it ties the label to the form field. :-)

noahmanger commented:
cc @LindsayYoung

Commit:
Merge branch 'develop' of github.com:18F/openFEC-web-app into feature/reports-pages

Commit:
Reverting to old range style

Commit:
Changing committee ID filter name

Commit:
Passing a single context object

Simplifies the context variables sent to the template by passing an
OrderedDict that is then converted to JSON, rather than setting
individual variables manually. h/t @ccostino

Commit:
Renaming template variable

Fixing a conflict between the context object passed to the template and
the built in context function

Commit:
Merge branch 'develop' of github.com:18F/openFEC-web-app into feature/reports-pages

Conflicts:
	openfecwebapp/constants.py
	openfecwebapp/templates/datatable.html
	openfecwebapp/templates/partials/disbursements-filter.html
	static/js/modules/columns.js
	static/js/pages/filings.js

Commit:
Removing old filters

Commit:
Removing pages column

noahmanger commented:
Ready for review. @xtine and @jontours can you check this out?

Commit:
Removing unnecessary underscore require

jontours commented:
@noahmanger I was able to test the web-app locally with my PR for this deployed locally, all looks good to me.

xtine commented on openfecwebapp/templates/partials/independent-expenditures-filter.html:
```diff
@@ -30,11 +31,8 @@
-    {{ text.field('min_amount', 'Minimum expenditure amount', {'data-suffix': 'or more', 'data-inputmask': '"alias": "currency", "rightAlign": false, "clearMaskOnLostFocus": true' }) }}
-    {{ text.field('max_amount', 'Maximum expenditure amount', {'data-suffix': 'or less', 'data-inputmask': '"alias": "currency", "rightAlign": false, "clearMaskOnLostFocus": true'}) }}
-
+    {# {% include 'partials/filters/ie-report-types.html' %} #}
```
Why is this still commented?

xtine commented:
@noahmanger: just had a question about a comment just for curiosity but otherwise looks good. 👍

Commit:
Removing commented report page

cc @xtine

noahmanger commented:
Fixed! Thanks.
